src/components/ChatEntry.js

```diff
@@ -1,22 +1,36 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
+const ChatEntry = ({ id, sender, body, timeStamp, liked, updateLike }) => {
+  const senderName =
+    sender === 'Vladimir' ? 'chat-entry local' : 'chat-entry remote';
+
+  const likeButtonContent = liked ? '❤️' : '🤍';
 
-const ChatEntry = (props) => {
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={senderName}>
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <p className="entry-time">
+          <TimeStamp time={timeStamp} />
+        </p>
+        <button className="like" onClick={() => updateLike(id, !liked)}>
+          {likeButtonContent}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  sender: PropTypes.string.isRequired,
```

Comment:
I'd recommend including the `id` prop in your prop types as well as making the `liked` and `updateLike` props required.